test/helpers/TestFileLoader.ts

```diff
@@ -0,0 +1,57 @@
+import fs from 'fs';
+import path from 'path';
+
+export namespace TestFileLoader {
+  export type FilterOption = { exclude?: string[]; include?: string[] };
+
+  const filter = (files: string[], options: FilterOption): string[] => {
+    return files.filter((file) => {
+      if (options.include) return isIncluded();
+      if (options.exclude) return !isExcluded();
+      return true;
+
+      function isExcluded() {
+        return options.exclude?.some((exPath) => file.includes(exPath));
+      }
+
+      function isIncluded() {
+        return options.include?.some((incPath) => file.includes(incPath));
+      }
+    });
+  };
+
+  const traverse = (dir: string, files: string[]) => {
+    const entries = fs.readdirSync(dir, { withFileTypes: true });
+
+    for (const entry of entries) {
+      if (isIgnoredEntry(entry)) continue;
+
+      const fullPath = path.join(dir, entry.name);
+
+      if (isTestDirectory(entry)) {
+        traverse(fullPath, files);
+      } else if (isTestFile(entry)) {
```

Comment:
well, In my opinion, there is always a possibility that the test code for any folder is `index.js`. Let's remove the dir from the [`load`](https://github.com/kakasoo/DeepStrictTypes/pull/8/files#diff-b573173d1826a17565e98b2f632d2a64798b736b8b33816b03e39b806164472bR51) function as a factor and keep it constant so that it only starts in the `test` folder. Then we can exclude only `index.js` of the outermost root from the test.

Reply:
I understand your opinion. I will modify the code to remove the root test folder from the test scope